src/Microsoft.Extensions.Configuration.AzureAppConfiguration/IKeyValueAdapter.cs

```diff
@@ -18,5 +18,7 @@ internal interface IKeyValueAdapter
         void InvalidateCache(ConfigurationSetting setting = null);
 
         bool NeedsRefresh();
+
+        void ResetState();
```

Comment:
I think you're right that at least something has to change here - having the new method seems weird since it's just for this one case. I think a rename works based on where we're calling `InvalidateCache` now and where the state needs to reset for the `FeatureManagementKeyValueAdapter` anyway.